flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/ChangelogStorageMetricGroup.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.changelog.fs;
+
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.Gauge;
+import org.apache.flink.metrics.Histogram;
+import org.apache.flink.runtime.metrics.DescriptiveStatisticsHistogram;
+import org.apache.flink.runtime.metrics.groups.ProxyMetricGroup;
+import org.apache.flink.runtime.metrics.groups.TaskManagerJobMetricGroup;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * Metrics related to the Changelog Storage used by the Changelog State Backend. Thread-safety is
+ * required because it is used by multiple uploader threads.
+ */
+@ThreadSafe
+public class ChangelogStorageMetricGroup extends ProxyMetricGroup<TaskManagerJobMetricGroup> {
+    private static final int WINDOW_SIZE = 1000;
+
+    private final Counter uploadsCounter;
+    private final Counter uploadFailures;
+    private final Histogram logsPerUpload;
+    private final Histogram uploadSizes;
+    private final Histogram uploadLatencies;
+    private final Histogram attemptsPerUpload;
+
+    public ChangelogStorageMetricGroup(TaskManagerJobMetricGroup parent) {
```

Comment:
This could work against the plain `MetricGroup` interface, correct? It would be good to reduce their usage to a minimum to ease maintenance.

Reply:
Done in 643a25496b3c7ccce9f7c796a095fe5e5be83400 (extracted PR).